proto/api.go

```diff
@@ -51,41 +52,187 @@ const (
 	isWrite                // write cmds go through raft and must be proposed on leader
 	isTxnWrite             // txn write cmds start heartbeat and are marked for intent resolution
 	isRange                // range commands may span multiple keys
+	isReverse              // reverse commands traverse ranges in descending direction
 )
 
 // IsAdmin returns true if the request requires admin permissions.
 func IsAdmin(args Request) bool {
+	if br, ok := args.(*BatchRequest); ok {
```

Comment:
You can also just post your WIP. That's what I'm doing here, and it's mostly for you to see. It'd be good if I could do the same. I've checked the active branches, but didn't see one?